src/components/StampOverlay/StampOverlay.js

```diff
@@ -108,6 +114,32 @@ class StampOverlay extends React.Component {
     this.setState({ isStampSelected: true });
   }
 
+  getCustomRubberStamps = async isLanChanged => {
+    if (!this.state.customAnnotations.length || isLanChanged) {
+      const annotations = await this.stampTool.getCustomStampAnnotations();
+      const previews = await Promise.all(
+        annotations.map(annotation => {
+          const text = this.props.t(`rubberStamp.${annotation['Icon']}`);
+
+          const options = {
+            canvasWidth,
+            canvasHeight,
+            text,
+          };
+
+          return this.stampTool.getPreview(annotation, options);
+        }),
+      );
+
+      const customAnnotations = annotations.map(annotation => ({
+        annotation,
+        imgSrc: annotation['ImageData'],
+      }));
+
+      this.setState({ customAnnotations });
```

Comment:
In the `getDefaultRubberStamps` function we have `this.setState({ defaultAnnotations, language: this.props.i18n.language });`, can you explain a little bit here about why we don't need set the state for `language` in this function?

Actually do you need to store `language` on the state? I don't see it being used in the render function.

Reply:
Here text for a dynamic stamp is entered by the user, and the user can enter it in any language, so we don't need to translate.

The this.state.language is used in componentDidUpdate in order to check if language changed at any time.